kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerLocator.java

```diff
@@ -15,19 +15,37 @@
 
 package org.kie.server.services.impl;
 
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.services.impl.storage.cloud.KieServerStateCloudRepository;
+
 public class KieServerLocator {
 
-    private static final KieServerImpl INSTANCE = new KieServerImpl();
-    
+    private static final KieServerImpl INSTANCE;
+    private static String strategy = System.getProperty(KieServerConstants.KIE_SERVER_STARTUP_STRATEGY);
+
     static {
+        if (strategy == null || strategy.length() == 0) {
+            strategy = System.getenv("KIE_SERVER_STARTUP_STRATEGY");
+
+            if (strategy == null || strategy.length() == 0) {
```

Comment:
Why check this? Line 37 should be fine with a null strategy and behave as before, and StartupStrategyProvider also defaults to this when null.

Reply:
Exist code does NOT have a consistency way of resolving StartupStrategy configuration. (See this line of code)
https://github.com/rhtevan/droolsjbpm-integration/blob/c03365c9cc4895c8a269edbbd3764cfa19321774/kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/StartupStrategyProvider.java#L35

It checks '-D' property **only** and default to ControllerBasedStartupStrategy. We'd like to fix this by implementing a consist configuration parameter resolution process, which is first to check '-D' java property, then fall back to system environment variable, at last set the missing property with a default value.